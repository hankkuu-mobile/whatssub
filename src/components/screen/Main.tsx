import { NavigationScreenProp, NavigationStateRoute } from 'react-navigation';
import React, { Component } from 'react';
import {
  Text,
  TouchableOpacity,
  View,
} from 'react-native';

import styled from 'styled-components/native';

const Container = styled.View`
  flex: 1;
  background-color: transparent;
  flex-direction: row;
  align-items: center;
  justify-content: center;
`;

const StyledText = styled.Text`
  font-size: 16;
  color: blue;
`;

interface Props {
  navigation?: NavigationScreenProp<any, any>;
}

function Page(props: Props) {
  return (
    <Container>
      <TouchableOpacity
        onPress={() => {
          props.navigation.navigate('CustomSubsRegister');
        }}
      >
        <StyledText testID='myText'>CustomSubsRegister screen</StyledText>
      </TouchableOpacity>
    </Container>
  );
}

export default Page;
