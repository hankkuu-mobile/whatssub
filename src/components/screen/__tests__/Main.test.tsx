import 'react-native';
import * as React from 'react';
import {
  RenderResult,
  act,
  fireEvent,
  render,
} from '@testing-library/react-native';
import Main from '../Main';

import renderer from 'react-test-renderer';

const createTestProps = (props: Object) => ({
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
  },
  ...props,
});

let props: any;
let component: React.ReactElement;
let testing: RenderResult;

describe('[Main]', () => {
  beforeAll(() => {
    props = createTestProps({});
  });

  it('renders without crashing', () => {
    const rendered: renderer.ReactTestRendererJSON =
      renderer.create(<Main />).toJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });
});

describe('[Main] Interaction', () => {
  beforeEach(() => {
    props = createTestProps({});
    component = <Main {...props} />;
    testing = render(component);
  });

  it('should render [Text] with value "myText"', () => {
    const textInstance = testing.getByTestId('myText');
    expect(textInstance.props.children).toEqual('CustomSubsRegister screen');
  });
});
