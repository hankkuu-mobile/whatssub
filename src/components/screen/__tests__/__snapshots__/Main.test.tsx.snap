// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Main] renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "flexBasis": 0,
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "blue",
            "fontSize": 16,
          },
        ]
      }
      testID="myText"
    >
      CustomSubsRegister screen
    </Text>
  </View>
</View>
`;
